### PR TITLE
Add headless_mode_will_redirect filter 

### LINKED ISSUE #14: [Feature] Filter to make it possible to do something else instead of
What if I wanted WordPress to respond to the HTTP request and spit out the HTML for my React app or whatever? Client-side routing, and same-origin requests is what's happening.

### PR DESCRIPTION
**Pull request by Shelob9**
Fixes #14

## What Has Changed

New filter so I think I need so I can use WordPress/ PHP to respond to HTTP requests that are not for REST API, etc with a React app or whatever.

### CODE REVIEW

**Comment by Benunc** [comment]
This PR looks simple enough, but i want to make sure I fully understand what's going on here, and how to test it before I merge it. 

Essentially the new filter defaults to 'true' and so unless a something is specifically filtering that value, all requests that make it to that point will redirect. To test it, I should add a code snippet to a site that filters the value of the new filter to `false` and then attempt to load the front end. 

I should also test by NOT filtering that value to ensure that it redirects correctly. 

@Shelob9 let me know if I am missing anything there.

File: headless-mode.php [hunk 1]
```diff
@@ -116,9 +116,19 @@ function headless_mode_disable_front_end() {
...
-		exit;
+		
+		/**
+		 * We should redirect. But will we?
+		 *
+		 * This filter allows you to do something else, when a redirect normally would occur.
+		 *
+		 * @param bool $will_redirct If truthy redirect will happen. If not, it will not. Do your own thing. Be you.
+		 */
+		if( apply_filters( 'headless_mode_will_redirect', true, $new_url ) ){
```
Comment (Benunc) [inline headless-mode.php:127]: what if this value is not set/filtered? I'm getting a white screen (nothing in debug or query monitor) for the case of not having the main constant `HEADLESS_MODE_CLIENT_URL` set, using this code.

**Review by Benunc — CHANGES_REQUESTED** [review]
Here's what I did to test:

1. set main `HEADLESS_MODE_CLIENT_URL` constant. This worked as expected.
2. wrote a custom function that returned "false" for the new `headless_mode_will_redirect` filter, no matter what. This worked as expected. 
3. Unset/commented out the main constant from step one, and removed my custom function from step 2. (simulating having installed the plugin but not set the constant yet.) This caused a whitescreen for any logged-out visitor. 
4. Re-enabled the function from step 2. This caused the redirect to stop working (because the value is false for the new filter) and made the white screen go away.

File: headless-mode.php (hunk 1 above)
Comment (Shelob9) [reply headless-mode.php:127]: If that returns false, headless mode is disabled. You would return false when you wanted WordPress theme to render the page.

Comment (Benunc) [reply headless-mode.php:127]: This appears to be unrelated to this PR, as I am getting a bug in a fresh install of the repo-version of the site. I am going to open a new issue with steps to replicate, and merge this one.

**Review by Benunc — COMMENTED** [review]
looks

**Pull request merged**